### PR TITLE
Add <IfLocale> component

### PR DESCRIPTION
**Pull request by Ken-Lo**
RE: https://github.com/gocardless/splash-pages/issues/168

This PR adds a `<IfLocale>` component. You can conditionally show information based on the local properties in each country (e.g. based on the domestic DD scheme or whether instant sign-up is available in the country).

```jsx
<IfLocale domesticScheme='bacs'>
  Stuff about Bacs Direct Debit
</IfLocale>
<IfLocale domesticScheme='sepa'>
  Stuff about SEPA Direct Debit
</IfLocale>
<IfLocale domesticScheme='bankgirot'>
  Stuff about Bg Autogiro
</IfLocale>

<IfLocale hasInstantSignup>
  Signup now in 5 minutes
</IfLocale>
<IfLocale hasInstantSignup={false}>
  Contact Sales
</IfLocale>

<IfLocale hasPercentagePricing>
  % pricing is available
</IfLocale>
<IfLocale hasPercentagePricing={false}>
  % pricing is not available
</IfLocale>
```

In order to make it work, the `country_properties` message needs to be set for each locale in the app/messages/ folder. e.g.

```js
  country: 'United Kingdom',
  country_properties: {
    local_scheme: 'bacs',
    has_instant_signup: true,
    has_percentage_pricing: true,
  },
```

```js
  country: 'Belgique',
  country_properties: {
    local_scheme: 'sepa',
    has_instant_signup: false,
    has_percentage_pricing: true,
  },
```

### CODE REVIEW

**Comment by Ken-Lo** [comment]
@harrison Would be good to get your thoughts on this approach.

Also thinking about possibly generalising this component to `<IfLocal>` and allowing it to work on other country-based properties (e.g. whether immediate sign-up is available, whether Basic pricing is available)

File: app/components/if-local-scheme/if-local-scheme.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+function getLocalesWithLocalScheme(selectedScheme, availableLocales, countryProperties) {
+  var localesWithScheme = [];
+
+  for (var locale in countryProperties) {
+    if (countryProperties[locale].local_scheme === selectedScheme) {
+      localesWithScheme.push(locale);
+    }
+  }
+
+  return localesWithScheme;
```
Comment (jackfranklin) [inline app/components/if-local-scheme/if-local-scheme.js]: you could use `filter` from LoDash here:

```js
return _.filter(countryProperties, function(locale) {
  return locale.local_scheme === selectedScheme;
});
```

**Comment by jackfranklin** [comment]
@Ken-Lo welcome to the frontend team! ;) This looks really awesome. I'll let @harrison review other than the one thing I spotted (happy to chat through further) but just wanted to say that this seems like a really nice, intuitive improvement that will help a lot :+1:

File: app/components/if-local-scheme/if-local-scheme.js (hunk 1 above)
Comment (harrison) [inline app/components/if-local-scheme/if-local-scheme.js]: :+1: 
but think you will need to return the object key:
```
return Object.keys(countryProperties).filter(function(locale) {
  return countryProperties[locale].local_scheme === selectedScheme;
});
```

File: app/components/if-local-scheme/if-local-scheme.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    children: React.PropTypes.node.isRequired,
+  }
+
+  static contextTypes = {
+    availableLocales: React.PropTypes.array.isRequired,
+    countryProperties: React.PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { availableLocales, countryProperties } = this.context;
```
Comment (harrison) [inline app/components/if-local-scheme/if-local-scheme.js]: `availableLocales` is unused

File: app/components/if-local-scheme/if-local-scheme.js (hunk 1 above)
Comment (harrison) [inline app/components/if-local-scheme/if-local-scheme.js]: Wait do you need this?

Could you do this in render?

```js
const schemes = getMessage(messages, 'country_properties');

 if (schemes. local_scheme !== this.props.name) { return null; }
return React.createElement('span', this.props, this.props.children);
```

Comment (harrison) [inline app/components/if-local-scheme/if-local-scheme.js]: think you could then remove `countryProperties`

Comment (Ken-Lo) [inline app/components/if-local-scheme/if-local-scheme.js]: Nice one @jackfranklin, I knew there must have been a better way of selecting items from an array (didn't find anything in the MDN docs). Thanks for the info about lodash!

@harrison Makes sense, if we aren't wrapping `<Translation>`, then we wouldn't need to know the properties of other countries. Will move to render and send you a v2 :)

**Comment by Ken-Lo** [comment]
@harrison Just updated this:

* Moved everything to render() so no need to pass countryProperties in the context
* Generalised to `<IfLocale>` so we can also do this on other country properties
* Added some tests
* Updated a couple of new pages to use the `<IfLocale>` tag

Would be good to get your thoughts!

File: app/components/if-locale/if-locale.spec.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          <TestIfLocale hasPercentagePricing>
+            foo
+          </TestIfLocale>
+        );
+
+        expect(renderText(component)).toEqual(null);
+      });
+    });
+  });
+
```
Comment (harrison) [inline app/components/if-locale/if-locale.spec.js]: :scissors: empty lines

File: app/components/if-locale/if-locale.spec.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  if (elements.length === 0) {
+    return null;
+  } else if (elements.length === 1) {
+    const element = ReactTestUtils.findRenderedDOMComponentWithTag(result, 'span');
+    return element.getDOMNode().textContent;
+  }
+}
+
+describe('IfLocale Component', () => {
+
```
Comment (harrison) [inline app/components/if-locale/if-locale.spec.js]: :scissors: empty lines

**Comment by harrison** [comment]
Nice work! :+1: minor comments

**Comment by harrison** [comment]
Closes #168

**Comment by Ken-Lo** [comment]
@harrison Thanks! Merging this now, will tidy up the FAQ & other pages to use `<IfLocale>` in a separate pull request.

**Pull request merged**